src/plugins/cosmos/components/modals/Staking/Staking.tsx

```diff
@@ -41,32 +43,65 @@ import { UnstakeConfirm } from './views/UnstakeConfirm'
 const StakingModalContent = ({ assetId, validatorAddress }: StakingModalProps) => {
   const location = useLocation()
   const translate = useTranslate()
+  const [steps, setSteps] = useState<StepConfig[]>()
+  const [isOverview, setIsOverview] = useState<match<{}> | null>()
+  const [isClaim, setIsClaim] = useState<match<{}> | null>()
+  const [isStake, setIsStake] = useState<match<{}> | null>()
+  const [isUnstake, setIsUnstake] = useState<match<{}> | null>()
 
-  const isOverview = matchPath(location.pathname, {
-    path: StakeRoutes.Overview,
-    exact: true,
-  })
+  useEffect(() => {
+    setIsOverview(
+      matchPath(location.pathname, {
+        path: StakeRoutes.Overview,
+        exact: true,
+      }),
+    )
 
-  const isClaim = matchPath(location.pathname, {
-    path: [ClaimPath.Confirm, ClaimPath.Broadcast],
-    exact: true,
-  })
+    setIsClaim(
+      matchPath(location.pathname, {
+        path: [ClaimPath.Confirm, ClaimPath.Broadcast],
+        exact: true,
+      }),
+    )
 
-  const isStake = matchPath(location.pathname, {
-    path: [StakeRoutes.Stake, StakingPath.Confirm, StakingPath.Broadcast],
-    exact: true,
-  })
+    setIsStake(
+      matchPath(location.pathname, {
+        path: [StakeRoutes.Stake, StakingPath.Confirm, StakingPath.Broadcast],
+        exact: true,
+      }),
+    )
 
-  const isUnstake = matchPath(location.pathname, {
-    path: [StakeRoutes.Unstake, UnstakingPath.Confirm, UnstakingPath.Broadcast],
-    exact: true,
-  })
+    setIsUnstake(
+      matchPath(location.pathname, {
+        path: [StakeRoutes.Unstake, UnstakingPath.Confirm, UnstakingPath.Broadcast],
+        exact: true,
+      }),
+    )
+  }, [location.pathname])
```

Comment:
That's making things even more verbose than they were already.
Is there another way we could write this so we don't have to call all these state setters on this `useEffect`?
What if we use `<Route />`s  instead of `matchPatch`s, would it make things cleaner?

Reply:
We could by using `isUnstake?.isExact` as `dependencies` of the `setSteps` useEffect instead of the object directly, wdyt?

Reply:
Since we're going to eventually refactor/unify the routing of Foxy/DeFi/Cosmos, let's stamp this PR and resolve this.